stl/src/atomic_wait.cpp

```diff
@@ -0,0 +1,311 @@
+        if (_Sync_module != nullptr) {
+            const auto _Wait_on_address =
+                reinterpret_cast<decltype(&::WaitOnAddress)>(GetProcAddress(_Sync_module, "WaitOnAddress"));
+            const auto _Wake_by_address_single =
+                reinterpret_cast<decltype(&::WakeByAddressSingle)>(GetProcAddress(_Sync_module, "WakeByAddressSingle"));
+            const auto _Wake_by_address_all =
+                reinterpret_cast<decltype(&::WakeByAddressAll)>(GetProcAddress(_Sync_module, "WakeByAddressAll"));
+
+            if (_Wait_on_address != nullptr && _Wake_by_address_single != nullptr && _Wake_by_address_all != nullptr) {
+                _Wait_functions._Pfn_WaitOnAddress.store(_Wait_on_address, _STD memory_order_relaxed);
+                _Wait_functions._Pfn_WakeByAddressSingle.store(_Wake_by_address_single, _STD memory_order_relaxed);
+                _Wait_functions._Pfn_WakeByAddressAll.store(_Wake_by_address_all, _STD memory_order_relaxed);
+                _Level = __std_atomic_api_level::__has_wait_on_address;
+            }
+        }
+
+        // for __has_srwlock, relaxed would have been enough, not distinguishing for consistency
+        _Wait_functions._Api_level.store(_Level, _STD memory_order_release);
+        return _Level;
+    }
+
+    [[nodiscard]] __std_atomic_api_level _Acquire_wait_functions() noexcept {
+        auto _Level = _Wait_functions._Api_level.load(_STD memory_order_acquire);
+        if (_Level <= __std_atomic_api_level::__detecting) {
+            _Level = _Init_wait_functions(_Level);
+        }
+
+        return _Level;
+    }
+
+    [[nodiscard]] BOOL __crtWaitOnAddress(
+        volatile VOID* Address, PVOID CompareAddress, SIZE_T AddressSize, DWORD dwMilliseconds) {
+        const auto _Wait_on_address = _Wait_functions._Pfn_WaitOnAddress.load(_STD memory_order_relaxed);
+        return _Wait_on_address(Address, CompareAddress, AddressSize, dwMilliseconds);
+    }
+
+    VOID __crtWakeByAddressSingle(PVOID Address) {
+        const auto _Wake_by_address_single = _Wait_functions._Pfn_WakeByAddressSingle.load(_STD memory_order_relaxed);
+        _Wake_by_address_single(Address);
+    }
+
+    VOID __crtWakeByAddressAll(PVOID Address) {
+        const auto _Wake_by_address_all = _Wait_functions._Pfn_WakeByAddressAll.load(_STD memory_order_relaxed);
+        _Wake_by_address_all(Address);
+    }
+#endif // _ATOMIC_WAIT_ON_ADDRESS_STATICALLY_AVAILABLE
+
+} // unnamed namespace
+
+_EXTERN_C
+int __stdcall __std_atomic_wait_direct(
+    const void* _Storage, const void* _Comparand, const size_t _Size, const unsigned long _Remaining_timeout) noexcept {
+#if _ATOMIC_WAIT_ON_ADDRESS_STATICALLY_AVAILABLE == 0
+    if (_Acquire_wait_functions() < __std_atomic_api_level::__has_wait_on_address) {
+        return __std_atomic_wait_indirect(_Storage, _Comparand, _Size, _Remaining_timeout);
+    }
+#endif // _ATOMIC_WAIT_ON_ADDRESS_STATICALLY_AVAILABLE == 0
+
+    BOOL result = __crtWaitOnAddress(

```

Comment:
I observe that this file has been fairly consistent about using `_Ugly` identifiers for local variables, except for `result` here. I recommend being consistent within a file. (The case for using `_Ugly` in `src` is that it's easier to not switch styles, and easier to move code to/from `inc` including declarations; the case for using `non_ugly` in `src` is that it's easier to read. Regardless, mixing `_Ugly` and `non_ugly` is the worst of both worlds.)